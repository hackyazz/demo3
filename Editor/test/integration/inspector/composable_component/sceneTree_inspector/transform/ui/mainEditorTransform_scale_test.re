open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open MainEditorTransform;

let _ =
  describe("MainEditorTransform scale", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    MainEditorTransformTestTool.transformBaseTest(
      sandbox,
      "test change scale value",
      (1., TransformUtils.getTransformScaleData),
      (
        MainEditorTransformTool.changeScaleX,
        MainEditorTransformTool.changeScaleY,
        MainEditorTransformTool.changeScaleZ,
      ),
    );
  });