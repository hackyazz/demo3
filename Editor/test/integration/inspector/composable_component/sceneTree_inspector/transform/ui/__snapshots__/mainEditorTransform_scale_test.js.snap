// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorTransform scale test change scale value changeX should set current gameObject->transform->value x set x value to floatInput 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="-10.1213"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
